testsuite/forge/src/backend/k8s/stateful_set.rs

```diff
@@ -0,0 +1,272 @@
+// Copyright (c) Aptos
+// SPDX-License-Identifier: Apache-2.0
+
+use crate::{Result, KUBECTL_BIN};
+use std::{process::Command, time::Duration};
+
+use anyhow::bail;
+use aptos_retrier::ExponentWithLimitDelay;
+use k8s_openapi::api::{apps::v1::StatefulSet, core::v1::Pod};
+use once_cell::sync::Lazy;
+
+use again::RetryPolicy;
+use aptos_logger::info;
+use kube::{
+    api::{Api, Meta, Patch, PatchParams},
+    client::Client as K8sClient,
+};
+use thiserror::Error;
+
+use crate::create_k8s_client;
+
+// retry every 10 seconds for 60 times, for up to 10 minutes
+// we want a fixed retry policy here to be as fast as possible
+static RETRY_POLICY: Lazy<RetryPolicy> =
```

Comment:
Do we really need a retry for up to 10 minutes here? What operation requires such long timeout?

Reply:
For a cluster that's spinning up from idle, we need to wait for the machines to be provisioned as well. Let me try splitting out the timeouts

Reply:
Ok changed it to exponential backoff up to 20 min for when we're first provisioning the namespace, and then fixed backoff for when we're rebooting an individual node